### PR TITLE
Rmpipelib

### PR DESCRIPTION
**Pull request by wkhunt**

### CODE REVIEW

File: JenkinsFile [hunk 1]
```diff
@@ -1,82 +1,238 @@
...
+        stage('Phase One Integration Tests') {
+          if(!fileExists("integration-tests")) {
+            sh "mkdir -p integration-tests"
+          }
+          // Clone Test Repository
+          withCredentials([file(credentialsId: "${POSTMAN_SECRET_FILE}", variable: 'POSTMAN_FILE')]) {
+            withEnv(["PATH+=${root}/integration-tests/node_modules/newman/bin:${nodejs}/bin", "PCF_SPACE=${env.PHASE_ONE_PCF_SPACE}", "HOME=${WORKSPACE}"]) {
+              dir ("integration-tests") {
+                if(env.GITLAB_CREDS) {
+                  git url: "https://github.com/venicegeo/pztest-integration.git", branch: "master", credentialsId: "${env.GITLAB_CREDS}"
```
Comment (benhosmer) [inline JenkinsFile]: Can we abstract this out?

File: JenkinsFile [hunk 2]
```diff
@@ -1,82 +1,238 @@
...
+                } else {
+                  git url: "https://github.com/venicegeo/pztest-integration.git", branch: "master"
+                }
+                sh "npm install newman@2"
```
Comment (benhosmer) [inline JenkinsFile]: You might run in to issues doing this like that.

You probably want to do something like this instead:

```
   def nodejs = tool 'NodeJS_6'
    def root = pwd()
    withEnv(["PATH+=${nodejs}/bin", "NPM_CONFIG_CACHE=${root}/.npmcache", "HOME=${WORKSPACE}"]) {
        sh "mkdir -p ${root}/.npmcache"
        sh "npm install newman"
        sh "pwd"
        sh "ls -al"
        sh "ls -al ${root}/.npmcache/newman/3.6.1/package/"
        sh "node -v"
        sh "npm -v"
        sh "newman -v"
    }
```

File: JenkinsFile [hunk 3]
```diff
@@ -1,82 +1,238 @@
...
     }
-    mavenPush()
-  }
-
-  stage('CI Deploy') {
-    cfPush()
-    zap {
-      threadfixId = THREADFIX_ID
+    
+    if(params.SECENV) {
```
Comment (benhosmer) [inline JenkinsFile]: You can remove this entire stage if you don't have any special configs you need to add for UC.

**Review by benhosmer — CHANGES_REQUESTED** [review]

File: JenkinsFile (hunk 1 above)
Comment (wkhunt) [reply JenkinsFile]: If I do all the java ones should too then

Comment (go0ty) [reply JenkinsFile]: Do any of the integration tests actually test the logger?  If not, we should remove these steps.

File: JenkinsFile [hunk 4]
```diff
@@ -1,82 +1,238 @@
...
+      stage('Phase Two Integration Tests') {
+        withCredentials([file(credentialsId: "${POSTMAN_SECRET_FILE}", variable: 'POSTMAN_FILE')]) {
+          withEnv(["PATH+=${root}/integration-tests/node_modules/newman/bin:${nodejs}/bin", "PCF_SPACE=${env.PHASE_TWO_PCF_SPACE}", "HOME=${WORKSPACE}"]) {
+            dir ("integration-tests") {
+              sh "ci/blackbox.sh"
+            }
+          }
+        }
+      }
+    }
```
Comment (go0ty) [inline JenkinsFile]: Same thing as int environment tests. If the integration tests don't test pz-logger at all, then we don't need these steps.

Comment (wkhunt) [reply JenkinsFile]: James says logger isnt tested directly but he wouldnt remove this

File: JenkinsFile (hunk 1 above)
Comment (wkhunt) [reply JenkinsFile]: James said he would keep them

**Review by benhosmer — APPROVED** [review]

**Pull request merged**